docs/api-reference/types/aggregate-root.md

```diff
@@ -88,133 +172,478 @@ public long ExpectedVersion { get; set; }
+                events);
+                
+            // Update the expected version for next save
+            aggregate.ExpectedVersion += events.Length;
+            
+            // Publish events to event handlers
+            foreach (var @event in events)
+            {
+                _eventPublisher.Publish(@event);
+            }
+        }
+        catch (ConcurrencyException ex)
+        {
+            // Handle optimistic concurrency conflicts
+            throw new AggregateVersionException(
+                $"Aggregate {aggregate.Id} has been modified concurrently", 
+                ex);
+        }
+    }
+}
+```
+
+### Initialize
+
+A common pattern in Reactive Domain is to use an `Initialize` method instead of raising events directly in the constructor. This allows for more explicit validation and better control over the initialization process.
+
+```csharp
+public void Initialize(/* parameters */, ICorrelatedMessage source)
+{
+    // Validate parameters
+    // ...
+    
+    // Raise initialization event
+    RaiseEvent(MessageBuilder.From(source, () => new EntityCreated(/* ... */)));
+}
+```
+
+**Example**:
+```csharp
+public class Product : AggregateRoot
+{
+    private string _name;
+    private string _sku;
+    private decimal _price;
+    private bool _isActive;
+    
+    public Product(Guid id) : base(id)
+    {
+        Register<ProductCreated>(Apply);
+        Register<ProductPriceChanged>(Apply);
+        Register<ProductDeactivated>(Apply);
+    }
+    
+    public void Initialize(string name, string sku, decimal price, ICorrelatedMessage source)
+    {
+        // Validate business rules
+        if (string.IsNullOrEmpty(name))
+            throw new ArgumentException("Product name is required", nameof(name));
+            
+        if (string.IsNullOrEmpty(sku))
+            throw new ArgumentException("SKU is required", nameof(sku));
+            
+        if (price <= 0)
+            throw new ArgumentException("Price must be greater than zero", nameof(price));
+            
+        // Raise the event
+        RaiseEvent(MessageBuilder.From(source, () => new ProductCreated(
+            Id, name, sku, price, DateTime.UtcNow)));
+    }
+    
+    private void Apply(ProductCreated @event)
+    {
+        _name = @event.Name;
+        _sku = @event.Sku;
+        _price = @event.Price;
+        _isActive = true;
+    }
+}
+```
 
 ## Usage
 
-The `AggregateRoot` class is designed to be subclassed by domain aggregates. Subclasses should:
+The `AggregateRoot` class is designed to be subclassed by domain aggregates. Here's a step-by-step guide for implementing aggregates in Reactive Domain:
 
-1. Define private `Apply` methods for each event type
-2. Use the `RaiseEvent` method to record and apply events
-3. Define public methods that represent domain operations
+1. **Create a Class**: Create a new class that inherits from `AggregateRoot`
+2. **Define State**: Define private fields to hold the aggregate's state
```

Comment:
An important point here is that the state of the aggregate should be as minimal as possible. The only state an aggregate should have is whatever is needed for it to make future business decisions.